src/sapling/saplingscriptpubkeyman.h

```diff
@@ -45,8 +45,11 @@ class SaplingNoteData
     SaplingNoteData(const libzcash::SaplingIncomingViewingKey& _ivk) : ivk {_ivk}, nullifier() { }
     SaplingNoteData(const libzcash::SaplingIncomingViewingKey& _ivk, const uint256& n) : ivk {_ivk}, nullifier(n) { }
 
+    /* witnesses/ivk: only for own (received) outputs */
     std::list<SaplingWitness> witnesses;
-    libzcash::SaplingIncomingViewingKey ivk;
+    Optional<libzcash::SaplingIncomingViewingKey> ivk;
+    inline bool IsMyNote() const { return ivk != nullopt; }
```

Comment:
I know that this is purely semantics but.. this is a pretty large puzzle and incoming are outgoing notes are notes from the wallet. I think that would be better to rename it to `IsIncoming()`.

Reply:
Since SaplingNoteData stores the info only from shielded outputs,  I think it's clear and makes more sense: "my outputs" are outputs "received" by me (there is no SaplingNoteData stored for spent notes... so there would be no `IsOutgoing()`).